src/RA_AchievementSet.cpp

```diff
@@ -480,85 +450,83 @@ BOOL AchievementSet::LoadFromFile(ra::GameID nGameID)
-                    {
-                        Achievement& newAch = AddAchievement();
-                        newAch.Parse(AchievementsData[i]);
-                    }
+                    auto& newAch{ AddAchievement() };
+                    newAch.Parse(achData);
                 }
-
-                if (m_nSetType != Core)
+                else if ((nFlags == 5) and (m_nSetType == Unofficial))
                 {
-                    fclose(pFile);
-                    return TRUE;
+                    auto& newAch{ AddAchievement() };
+                    newAch.Parse(achData);
                 }
+                else
+                    continue;
+            }
 
-                const Value& LeaderboardsData = doc["Leaderboards"];
-                for (SizeType i = 0; i < LeaderboardsData.Size(); ++i)
-                {
-                    //"Leaderboards":[{"ID":"2","Mem":"STA:0xfe10=h0000_0xhf601=h0c_d0xhf601!=h0c_0xfff0=0_0xfffb=0::CAN:0xhfe13<d0xhfe13::SUB:0xf7cc!=0_d0xf7cc=0::VAL:0xhfe24*1_0xhfe25*60_0xhfe22*3600","Format":"TIME","Title":"Green Hill Act 1","Description":"Complete this act in the fastest time!"},
+            if (m_nSetType != Core)
+                return TRUE;
 
-                    auto& lbData = LeaderboardsData[i];
-                    RA_Leaderboard lb(lbData["ID"].GetUint());
+            const auto& LeaderboardsData{ doc["Leaderboards"] };
+            for (auto& lbData : LeaderboardsData.GetArray())
+            {
+                //"Leaderboards":[{"ID":"2","Mem":"STA:0xfe10=h0000_0xhf601=h0c_d0xhf601!=h0c_0xfff0=0_0xfffb=0::CAN:0xhfe13<d0xhfe13::SUB:0xf7cc!=0_d0xf7cc=0::VAL:0xhfe24*1_0xhfe25*60_0xhfe22*3600","Format":"TIME","Title":"Green Hill Act 1","Description":"Complete this act in the fastest time!"},
 
-                    lb.SetTitle(lbData["Title"].GetString());
-                    lb.SetDescription(lbData["Description"].GetString());
+                RA_Leaderboard lb{ lbData["ID"].GetUint() };
 
-                    auto nFormat = MemValue::ParseFormat(lbData["Format"].GetString());
-                    lb.ParseFromString(lbData["Mem"].GetString(), nFormat);
+                lb.SetTitle(lbData["Title"].GetString());
+                lb.SetDescription(lbData["Description"].GetString());
 
-                    g_LeaderboardManager.AddLeaderboard(lb);
-                }
-            }
-            else
-            {
-                fclose(pFile);
-                ASSERT(!"Could not parse file?!");
-                return FALSE;
-            }
+                auto nFormat{ MemValue::ParseFormat(lbData["Format"].GetString()) };
+                lb.ParseFromString(lbData["Mem"].GetString(), nFormat);
 
-            fclose(pFile);
+                g_LeaderboardManager.AddLeaderboard(lb);
+            }
 
             // calculate the total number of points for the core set, and pre-fetch badge images
-            unsigned int nTotalPoints = 0;
+            unsigned int nTotalPoints{};
```

Comment:
Sure thing, {} just means give me the default. Not a big deal to add 0U. But for (unsigned)short there is no known literal, but if it's that important it will "0".